storage/log_storage.go

```diff
@@ -102,9 +87,7 @@ type LogTreeTX interface {
 // ReadOnlyLogStorage represents a narrowed read-only view into a LogStorage.
 type ReadOnlyLogStorage interface {
 	DatabaseChecker
-
-	// Snapshot starts a read-only transaction not tied to any particular tree.
-	Snapshot(ctx context.Context) (ReadOnlyLogTX, error)
+	LogMetadata
```

Comment:
I don't think there's any mention of this interface, so what would you think of inlining the method here, and deleting `LogMetadata` entirely?

Reply:
SGTM. I'll do it in a follow-up PR if you don't mind?

Reply:
Absolutely, no problem.

Reply:
https://github.com/google/trillian/pull/2550